grafana/grafana
# Logs: use explicit frame-type-check

the dataframes returned by Loki for logs contain a special field named "labels".
in the generic logs-code, previously we just looked for a field named "labels", and if it was there, we assumed it's the thing with the labels in it. but there can be false positives, an another datasource might return data with a `labels` field that is not related to this.

to be more explicit, we should create a new frame-type value for this format. that's happening at https://github.com/grafana/grafana-plugin-sdk-go/pull/483 .

until that PR is merged, we will use a custom-field in the frame's meta to say that we have this format. this will fix the false-positive problem.

how to test:
1. make sure you have the `lokiBackendMode` feature-flag enabled
2. have a loki datasource, go to grafana-explore, run a loki logs query that returns data with labels
3. in the logs-visualization, open a log-line, and verify that there is a "Log labels" section, with content in it, not just a "Detected fields" section
4. now disable the `lokiBackendMOde` feature-flag, and repeat the test.
